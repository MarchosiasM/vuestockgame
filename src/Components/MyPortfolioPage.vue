<template>
  <div class="container">
    <div v-for="company in myStocks" :key="company.index">
      <my-portfolio-card
        :name="company.name"
        :count="company.count"
        :price="company.price"
        :index="company.index">

      </my-portfolio-card>
    </div>
  </div>
</template>

<script>
import MyPortfolioCard from './MyPortfolioCard.vue';
export default {
  components: {
    MyPortfolioCard
  },
  computed: {
    myStocks() {
      return this.$store.getters.fullPortfolioData;
    }
  }
};
</script>
