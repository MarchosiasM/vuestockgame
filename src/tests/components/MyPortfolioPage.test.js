import { mount } from '@vue/test-utils';
import MyPortfolioPage from '../../Components/MyPortfolioPage.vue';
import storeObj from './store/store';

const { store: impStore, localVue } = storeObj;

let store;

describe('MyPortfolioPage', () => {
  beforeEach(() => {
    store = Object.assign(impStore);
  });
  it('renders correctly', () => {
    const wrapper = mount(MyPortfolioPage, { store, localVue });
    expect(wrapper.element).toMatchSnapshot();
  });
  it('renders a daughter element for each company the player has', () => {
    const wrapper = mount(MyPortfolioPage, { store, localVue });
    const { personalPortfolio } = store.state;
    const stocksCards = wrapper.findAll('.my-portfolio-card');
    expect(stocksCards.wrappers).toHaveLength(Object.keys(personalPortfolio).length);
  });
});
