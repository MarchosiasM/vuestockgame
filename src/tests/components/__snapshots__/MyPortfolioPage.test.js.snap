// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPortfolioPage renders correctly 1`] = `
<div
  class="container"
>
  <div>
    <div
      class="my-portfolio-card col-md-6"
    >
      <p>
        Name: google || Count: 30 || Price: 130 
      </p>
       
      <div
        class="form-group"
      >
        <input
          type="text"
        />
         
        <button
          class="btn btn-primary"
        >
          Sell
        </button>
      </div>
    </div>
  </div>
  <div>
    <div
      class="my-portfolio-card col-md-6"
    >
      <p>
        Name: amd || Count: 350 || Price: 130 
      </p>
       
      <div
        class="form-group"
      >
        <input
          type="text"
        />
         
        <button
          class="btn btn-primary"
        >
          Sell
        </button>
      </div>
    </div>
  </div>
  <div>
    <div
      class="my-portfolio-card col-md-6"
    >
      <p>
        Name: tesla || Count: 50 || Price: 130 
      </p>
       
      <div
        class="form-group"
      >
        <input
          type="text"
        />
         
        <button
          class="btn btn-primary"
        >
          Sell
        </button>
      </div>
    </div>
  </div>
</div>
`;
